pkg/blockchain/ledger.go

```diff
@@ -123,8 +139,17 @@ func (l *Ledger) Update(f *Ledger, h *hub.Message, c chan *hub.Message) (err err
 		return
 	}
 
+	if len(l.protectedStoreKeys) > 0 && !slices.Contains(l.trustedPeerIDS, h.SenderID) {
+		for _, key := range l.protectedStoreKeys {
+			if !maps.Equal(l.blockchain.Last().Storage[key], block.Storage[key]) {
+				err = errors.Wrapf(err, "unauthorized attempt to write to protected bucket: %s", key)
+				return
+			}
+		}
+	}
+
 	l.Lock()
-	if block.Index > l.blockchain.Len() {
+	if l.skipVerify || block.Index > l.blockchain.Len() {
```

Comment:
I'm not sure - why do we need still skipVerify? otherwise changes looks good to me!

Reply:
As i mentioned before, we need to have nodes as passive listeners, and in that mode they listen only from trusted peers

> Yeah, either we need to have a reconciliation mechanism implemented, still having priority over trusted nodes vs. block validation Either we having a full passive mode, like i did, which is more simple to implement
> 
> We can still verify blocks for everyone, if we haven't defined the trusted nodes, so i would not remove that bits if i understand you correctly, because edgevpn still can be used as fully decentralized non-authoritarian VPN this way (yet, i see that later joined peers never go in sync that way, but for now that is needed only for healthchecks and other simple metadata)

Reply:
Sorry, I was just checking out again what's left here in this PR! 

The problem I have with ignoring the block index is that potentially you might have node broadcasting old chains and/or completely separated chains. This causes network splits.